pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1940,20 +1942,35 @@ func handleDelete(ctx *domainContext, key string, status *types.DomainStatus) {
 // returns domainID, PodUUID and error
 func DomainCreate(status types.DomainStatus) (int, string, error) {
 
-	var domainID int
-	var err error
-	podUUID := ""
+	var (
+		domainID int
+		err      error
+		podUUID  string
+	)
 
 	filename := xenCfgFilename(status.AppNum)
 	log.Infof("DomainCreate %s ... xenCfgFilename - %s\n", status.DomainName, filename)
 	if status.IsContainer {
 		// Use rkt tool
 		log.Infof("Using rkt tool ... ContainerImageID - %s\n", status.ContainerImageID)
-		aciFileName := filepath.Join(types.VerifiedAppImgDirname,
-			status.DiskStatusList[0].ImageID.String(),
-			status.ContainerImageID)
-		aciFileName = aciFileName + ".aci"
-		domainID, podUUID, err = rktRun(status.DomainName, status.ContainerImageID, filename, aciFileName)
+		tarFilename, err := utils.VerifiedImageFileLocation(status.IsContainer,
```

Comment:
I'm compating this call to VerifiedfImageFileLocation with the call on line 1364, and they feed in different arguments. Seems quite odd.

Reply:
I will admit quite openly that some of the path resolution had me confused for a while. 

For this one, in 1364, the second argument is part of a loop, so `ds = status.DiskStatusList[0]`, and thus `ds.ImageID.String()` is the same as `status.DiskStatusList[0].ImageID.String()`

For the third argument, I don't fully get it myself; I just took the arguments as they were there. :-(

Reply:
I see that VerifiedImageFileLocation makes different assumptions about the third arg depending on the IsContainer arg.